### ISSUE TITLE
[vue] process hangs due to babel parser error

### ISSUE THREAD

**Issue by lkho** [prettier/prettier#12112]
**Prettier 2.5.1**
[Playground1](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8AOlAARmoAmAlgG4nkWW11l0C0AZhAE4C8xEGTxka5IYTzTpqAPStGFeYtLzMuAnEYgANCAhYYNaAGdkoPL14QA7gAUrCcyjw5beAJ7n9AI154YADWcDAAyngYcAAy4nDIXG6mcH4BwaFhWIHiAObIMLwArikgcBi+cFRUldF4UDmFeDlwAGJ8GATG9cggeIUwEHogABYwGDgA6sM08KZZYHBhzjP0M549YKY+IOLJvDD2ATkdCUklAFamAB5huThwAIqFEPCnOMn6Wbx7PXTFQ1heOIYBMaFQYMNkAAOAAMnxsyQmASwPUBcD2dHi+gAjs94IdDC5eqYOFA4JVKkNeHBcTRqYcmickIl3iVkhgaPkimy7o88fFmWd9DA8L5QeDIUgAEzCgI0HC5ADCEAwTNKpgArENCskACqilwsj4gP5wACSUGqsDCYCBRgAgpawjBPPc3skAL4eoA)

**Input 1:**

```vue
<template>
  <div>
    <div v-for=" a in ">aaaaa</div>
  </div>
</template>
```

**Output 1:**

```vue
SyntaxError: Unexpected token (1:1)
> 1 |
    | ^
```

[Playground2](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeeBbADgGwIbwB8AOlAARmoAmAlgG4nkWW11l0C0AZhAE4C8xEGTxkaUIYTzTpqAPStGFeYtLzMuAnEYgANCAhYYNaAGdkoPL14QA7gAUrCcyjw5beAJ7n9AI154YADWcDAAyngYcAAy4nDIXG6mcH4BwaFhWIHiAObIMLwArikgcBi+cFRUldF4UDmFeDlwAGJ8GATG9cggeIUwEHogABYwGDgA6sM08KZZYHBhzjP0M549YKY+IOLJvDD2ATkdCUklAFamAB5huThwAIqFEPCnOMn6Wbx7PXTFQ1heOIYBMaFQYMNkAAOAAMnxsyQmASwPUBcD2dHi+gAjs94IdDC5eqYOFA4JVKkNeHBcTRqYcmickIl3iVkhgaPkimy7o88fFmWd9DA8L5QeDIUgAEzCgI0HC5ADCEAwTNKpgArENCskACqilwsj4gP5wACSUGqsDCYCBRgAgpawjBPPc3skAL4eoA)

**Input 2:**

```vue
<template>
  <div>
    <div v-for=" a in">aaaaa</div>
  </div>
</template>
```

**Output 2:**

```
TypeError: Cannot destructure property 'left' of '(intermediate value)(...)' as it is undefined.
    at printVueFor (https://prettier.io/lib/standalone.js:40:13897)
    at mz (https://prettier.io/lib/standalone.js:40:16762)
    at Object.yz [as embed] (https://prettier.io/lib/standalone.js:40:19230)
    at Object.printSubtree (https://prettier.io/lib/standalone.js:1:220684)
    at yD (https://prettier.io/lib/standalone.js:1:222198)
    at a (https://prettier.io/lib/standalone.js:1:221787)
    at s (https://prettier.io/lib/standalone.js:1:221610)
    at https://prettier.io/lib/standalone.js:40:9647
    at https://prettier.io/lib/standalone.js:1:219866
    at iD.each (https://prettier.io/lib/standalone.js:1:219698)
```

**Expected behavior:**

**Output 2** _should be_ the expected behavior of both **Input 1** and **Input 2** since both inputs is a vue syntax error missing the RHS of a `v-for`

**The current problem is**, while the playground here successfully return **Output 1** as an error, however, **Output 1** is thrown at the `babel-parser` level. When you try to run this **Input 1** yourself in your own CLI, with the default highlighting turned _on_, the parser error will bubbles to the top and go through the highlighting process by `codeFrameColumns (from @babel/code-frame)`, with an empty `text = ""` value.

https://github.com/prettier/prettier/blob/514046b3c70e6477cb69b4d36871d0d4c8c5e415/src/main/parser.js#L103

When you step into the implementation, the code loops infinitely inside the `tokenize*` generator function.

# TL;DR
So the current behavior is, when you run **Input 1** in a CLI terminal (i.e. save the code in a file e.g. `abc.vue` and run `$ prettier abc.vue`) with highlighting turned _on_, the process hangs, and CPU is at 100%.

**Comment by lkho** [comment]
https://github.com/babel/babel/pull/14165
